tests/test_modeling_tf_common.py

```diff
@@ -399,6 +408,58 @@ def test_pt_tf_model_equivalence(self):
             max_diff = np.amax(np.abs(tf_hidden_states - pt_hidden_states))
             self.assertLessEqual(max_diff, 4e-2)
 
+            has_labels = any(
+                x in tf_inputs_dict_maybe_with_labels for x in ["labels", "next_sentence_label", "start_positions"]
+            )
+            if has_labels:
+
+                with torch.no_grad():
+                    pto = pt_model(**pt_inputs_dict_maybe_with_labels)
+                tfo = tf_model(tf_inputs_dict_maybe_with_labels, training=False)
+
+                # Some models' output class don't have `loss` attribute despite `labels` is used.
+                tf_loss = getattr(tfo, "loss", None)
+                pt_loss = getattr(pto, "loss", None)
+
+                # Some models require extra condition to return loss. For example, `BertForPreTraining` requires both
+                # `labels` and `next_sentence_label`.
+                # Moreover, some PT models return loss while the corresponding TF/Flax models don't.
+                if tf_loss is not None and pt_loss is not None:
+
+                    tf_loss = tf.math.reduce_mean(tf_loss).numpy()
+                    pt_loss = pt_loss.numpy()
+
+                    tf_nans = np.copy(np.isnan(tf_loss))
+                    pt_nans = np.copy(np.isnan(pt_loss))
```

Comment:
Curiosity: any particular reason to use `np.copy()`?

Reply:
It's copied from other places inside the same method. I don't see the necessity though.

Reply:
(not checked, just guess)

probably it is because we modify the `hidden_states`, which **probably** will affect the results of `np.isnan` (depends on if this method return just a view or a new copied data )

```
            tf_nans = np.copy(np.isnan(tf_hidden_states))
            pt_nans = np.copy(np.isnan(pt_hidden_states))

            pt_hidden_states[tf_nans] = 0
            tf_hidden_states[tf_nans] = 0
```

Reply:
Good point! 

I believe it is returning a copy. If we run the commands below, we can see that the output doesn't change if we change the original input object with an in-place operation.

```python
>>> import numpy as np
>>> a = np.asarray([np.nan, 1, 2, 3, np.nan])
>>> b = np.isnan(a)
>>> b
array([True, False, False, False,  True])
>>> a.sort()
>>> a
array([ 1.,  2.,  3., nan, nan])
>>> b
array([True, False, False, False,  True])
```

That being said, it should be safe to remove the copy. If we get errors from removing the copy, then we can add this quirk as a comment.

Reply:
I agree, thanks for testing, @gante ! If you & @Rocketknight1 are OK, I can change it in this PR (although it is not this PR's original intention, but I'm fine with it).

Reply:
I removed `np.copy`, and the test is fine :)